src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,117 @@
+            preparedStatement.setObject(1, id);
+            ResultSet generatedKeys = preparedStatement.executeQuery();
+            if (generatedKeys.next()) {
+                String name = generatedKeys.getString(COLUMN_NAME);
+                String country = generatedKeys.getString(COLUMN_COUNTRY);
+                return Optional.of(new Manufacturer(id, name, country));
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't get manufacturer by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> manufacturers;
+        String request = "SELECT * FROM manufacturers WHERE is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement statement = connection.createStatement()) {
+            ResultSet resultSet = statement
+                    .executeQuery(request);
+            manufacturers = getManufacturers(resultSet);
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't operate with DB", e);
+        }
+        return manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String request = "UPDATE manufacturers "
+                + "SET name = ?, country = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement query = connection.prepareStatement(request)) {
+            query.setString(1, manufacturer.getName());
+            query.setString(2, manufacturer.getCountry());
+            query.setObject(3, manufacturer.getId());
+            query.executeUpdate();
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't update DB with " + manufacturer, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String request = "UPDATE manufacturers SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement query = connection.prepareStatement(request)) {
+            query.setObject(1, id);
+            return query.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't delete by id " + id, e);
+        }
+    }
+
+    private List<Manufacturer> getManufacturers(ResultSet resultSet) throws SQLException {
+        List<Manufacturer> manufacturerList = new ArrayList<>();
+        while (resultSet.next()) {
+            String name = resultSet.getString(COLUMN_NAME);
+            String country = resultSet.getString(COLUMN_COUNTRY);
+            Long id = resultSet.getObject(COLUMN_ID, Long.class);
+            manufacturerList.add(new Manufacturer(id, name, country));
+        }
+        return manufacturerList;
+    }
```

Comment:
```suggestion
    private Manufacturer getManufacturer(ResultSet resultSet) throws SQLException {
            String name = resultSet.getString(COLUMN_NAME);
            String country = resultSet.getString(COLUMN_COUNTRY);
            Long id = resultSet.getObject(COLUMN_ID, Long.class);
            return new Manufacturer(id, name, country);
    }
```
and then we can use this method in the get() and getAll() methods